device/globals/dialog_instance.gd

```diff
@@ -69,6 +71,26 @@ func finish():
 			anim.play("hide")
 	_queue_free()
 
+func clamped_position(dialog_pos):
+	var my_size = $"anchor/text".get_size()
+	var center_offset = my_size.x / 2
+
+	var dist_from_right = background_size.x - (dialog_pos.x + center_offset)
+	var dist_from_left = dialog_pos.x - center_offset
+	var dist_from_bottom = background_size.y - (dialog_pos.y + my_size.y)
+	var dist_from_top = dialog_pos.y - my_size.y
+
+	if dist_from_right < 0:
+		dialog_pos.x += dist_from_right
+	if dist_from_left < 0:
+		dialog_pos.x -= dist_from_left
+	if dist_from_bottom < 0:
+		dialog_pos.y += dist_from_bottom
+	if dist_from_top < 0:
+		dialog_pos.y -= dist_from_top
+
+	return dialog_pos
+
```

Comment:
I wish they could, but I'm not so certain because the action menu should be centered on the cursor click whereas dialog requires only the x-axis to be accounted for.

Tooltips are also horribly broken until there's a way to get the text-width size of the tooltip RichTextLabel so they can't be clamped except vertically.

So hopefully in the future!